flair/trie/replication.go

```diff
@@ -19,6 +20,8 @@ var log = logging.MustGetLogger()
 
 var serverDead = status.Errorf(codes.DeadlineExceeded, "Server is down")
 
+var rnd = rand.New(rand.NewSource(time.Now().UnixNano()))
```

Comment:
The `Rand` type isn't safe for concurrent use. This would be better as a member of the `Replicator` struct but I still don't think that's enough (you could get multiple calls going through the same replicator simultaneously)

Reply:
Cool, I’ve just made the intN function aliased in the package and override in tests, which should do the trick!